instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -319,15 +286,19 @@ private void afterConsumerResponse(
     }
   }
 
+  @SuppressWarnings({"rawtypes", "unchecked"})
   private void createConsumerSpan(
       Object message, ExecutionAttributes executionAttributes, SdkHttpResponse httpResponse) {
 
     io.opentelemetry.context.Context parentContext = io.opentelemetry.context.Context.root();
 
     if (messagingPropagator != null) {
+      // TODO: Dirty cast, only occurs due to partial removal of reflection.
       parentContext =
           SqsParentContext.ofMessageAttributes(
-              SqsMessageAccess.getMessageAttributes(message), messagingPropagator);
+              (Map<String, MessageAttributeValue>)
```

Comment:
If we are going to rely on generic signatures being exempt from muzzle checks we could use `Map<String, MessageAttributeValue>` as return type for `SqsMessageAccess.getMessageAttributes`, wdyt?